Clusteredverticalbar.js

```diff
@@ -0,0 +1,994 @@
+
+            d3.select(div).append('div')
+                .attr('class', 'sort_selection');
+
+            d3.select(div).append('div')
+                .attr('class', 'arrow-down');
+
+            var str = UTIL.createAlert($(div).attr('id'), _measure);
+            $(div).append(str);
+
+            var container = svg.append('g')
+                .attr('transform', 'translate(' + COMMON.PADDING + ', ' + COMMON.PADDING + ')');
+
+            $(document).on('click', 'svg', function (e) {
+                if ($("#myonoffswitch").prop('checked') == false) {
+                    var element = e.target
+                    if (element.tagName == "svg") {
+                        $('#Modal_' + $(div).attr('id') + ' .measure').val('')
+                        $('#Modal_' + $(div).attr('id') + ' .threshold').val('')
+                        $('#Modal_' + $(div).attr('id') + ' .measure').attr('disabled', false)
+                        $('#Modal_' + $(div).attr('id')).modal('toggle');
+                    }
+                }
+            })
+
+            $(document).on('click', '#Modal_' + $(div).attr('id') + ' .ThresholdSubmit', function (e) {
+                var newValue = $('#Modal_' + $(div).attr('id') + ' .threshold').val();
+                var obj = new Object()
+                obj.measure = $('#Modal_' + $(div).attr('id') + ' .measure').val()
+                obj.threshold = newValue;
+                threshold.push(obj);
+                $('#Modal_' + $(div).attr('id')).modal('toggle');
+            })
+
+            var legendWidth = 0,
+                legendHeight = 0;
+
+            plotWidth = parentWidth;
+            plotHeight = parentHeight;
+
+            var legendBreakCount;
+
+            if (_showLegend) {
+                var clusteredverticalbarLegend = LEGEND.bind(chart);
+
+                var result = clusteredverticalbarLegend(_legendData, container, {
+                    width: parentWidth,
+                    height: parentHeight,
+                    legendBreakCount: legendBreakCount
+                });
+
+                legendWidth = result.legendWidth;
+                legendHeight = result.legendHeight;
+                legendBreakCount = result.legendBreakCount;
+
+                switch (_legendPosition) {
+                    case 'top':
+                        plotHeight = parentHeight - legendHeight - axisLabelSpace;
+                        break;
+                    case 'bottom':
+                        plotHeight = parentHeight - legendHeight - axisLabelSpace * 2;
+                        break;
+                    case 'right':
+                    case 'left':
+                        plotWidth = parentWidth - legendWidth;
+                        break;
+                }
+            }
+
+            if (_tooltip) {
+                tooltip = d3.select(this.parentNode).select('#tooltip');
+            }
+            chart.drawViz = function (element) {
```

Comment:
Let's not create a public function drawViz on the chart module. We can create a private function or code block which is not accessible to the outsider